### PR TITLE
Refactor Mapbox logging

### PR DESCRIPTION
**Pull request by kiryldz**
### Summary of changes

Revisit the way SDK write logs. Any Mapbox log entry will have TAG=Mapbox, previous tag will now be appended to log text message. Additionally we mark logs coming directly from Android SDK (and not core part) with `maps-android`.

Assuming we had entry:
```
2022-04-12 15:29:59.987 10481-10569/com.mapbox.maps.testapp I/Mbgl-EGLConfigChooser: In emulator: false
```
it will look like this now:
```
2022-04-12 15:29:59.987 10481-10569/com.mapbox.maps.testapp I/Mapbox: [maps-android\Mbgl-EGLConfigChooser]: In emulator: false
```

This change allows to easily filter out log entries produced by each component of Maps SDK, e.g.:
```
2022-04-12 15:29:59.987 10481-10569/com.mapbox.maps.testapp I/Mapbox: [maps-android\Mbgl-EGLConfigChooser]: In emulator: false
2022-04-12 15:29:59.988 10481-10569/com.mapbox.maps.testapp I/Mapbox: [maps-android\Mbgl-EGLConfigChooser]: In emulator: false
2022-04-12 15:29:59.994 10481-10569/com.mapbox.maps.testapp I/Mapbox: [maps-android\Mbgl-EglCore]: EGLContext created, client version 2
2022-04-12 15:29:59.994 10481-10569/com.mapbox.maps.testapp I/Mapbox: [maps-core]: Using OpenGL render backend
2022-04-12 15:30:00.009 10481-10617/com.mapbox.maps.testapp I/Mapbox: [maps-core]: {MapboxRenderThr}[General]: GPU Identifier: Mali-G77
```

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [x] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [x] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/examples/SpaceStationLocationActivity.kt [hunk 1]
```diff
@@ -6,14 +6,10 @@ import android.os.Handler
...
 import androidx.appcompat.app.AppCompatActivity
-import com.mapbox.common.Logger
 import com.mapbox.geojson.Feature
 import com.mapbox.geojson.FeatureCollection
 import com.mapbox.geojson.Point
-import com.mapbox.maps.CameraOptions
-import com.mapbox.maps.MapView
-import com.mapbox.maps.MapboxMap
-import com.mapbox.maps.Style
+import com.mapbox.maps.*
```
Comment (alexander-kulikovskii) [inline app/src/main/java/com/mapbox/maps/testapp/examples/SpaceStationLocationActivity.kt:12]: can you return back import list?

File: app/src/main/java/com/mapbox/maps/testapp/examples/customlayer/CustomLayerActivity.kt [hunk 2]
```diff
@@ -4,12 +4,8 @@ import android.os.Bundle
 import android.view.Menu
 import android.view.MenuItem
 import androidx.appcompat.app.AppCompatActivity
-import com.mapbox.common.Logger
 import com.mapbox.geojson.Point
-import com.mapbox.maps.CameraOptions
-import com.mapbox.maps.LayerPosition
-import com.mapbox.maps.MapboxMap
-import com.mapbox.maps.Style
+import com.mapbox.maps.*
```
Comment (alexander-kulikovskii) [inline app/src/main/java/com/mapbox/maps/testapp/examples/customlayer/CustomLayerActivity.kt:8]: import

File: plugin-annotation/src/test/java/com/mapbox/maps/plugin/annotation/generated/CircleAnnotationManagerTest.kt [hunk 3]
```diff
@@ -12,11 +12,7 @@ import com.mapbox.common.Cancelable
 import com.mapbox.geojson.Feature
 import com.mapbox.geojson.FeatureCollection
 import com.mapbox.geojson.Point
-import com.mapbox.maps.LayerPosition
-import com.mapbox.maps.QueriedFeature
-import com.mapbox.maps.QueryFeaturesCallback
-import com.mapbox.maps.RenderedQueryGeometry
-import com.mapbox.maps.ScreenCoordinate
+import com.mapbox.maps.*
```
Comment (alexander-kulikovskii) [inline plugin-annotation/src/test/java/com/mapbox/maps/plugin/annotation/generated/CircleAnnotationManagerTest.kt:15]: import

File: sdk/src/test/java/com/mapbox/maps/renderer/MapboxRendererTest.kt [hunk 4]
```diff
@@ -3,11 +3,7 @@ package com.mapbox.maps.renderer
 import android.os.Handler
 import android.os.HandlerThread
 import android.os.Looper
-import com.mapbox.common.ShadowLogger
-import com.mapbox.maps.MapInterface
-import com.mapbox.maps.MapView
-import com.mapbox.maps.Size
-import com.mapbox.maps.Task
+import com.mapbox.maps.*
```
Comment (alexander-kulikovskii) [inline sdk/src/test/java/com/mapbox/maps/renderer/MapboxRendererTest.kt:6]: maybe we need rule for AS for wildcard imports?

Comment (kiryldz) [reply sdk/src/test/java/com/mapbox/maps/renderer/MapboxRendererTest.kt:6]: we don't have it now so I see no issue in using it, we're doing this in dozens of other places. Let's revisit that separately in dedicated PR if we decide on it, WDYT?

**Review by Chaoba — APPROVED** [review]

File: sdk/src/test/java/com/mapbox/maps/renderer/RenderHandlerThreadTest.kt [hunk 5]
```diff
@@ -24,6 +21,8 @@ class RenderHandlerThreadTest {
   @Before
   fun setUp() {
     renderHandlerThread = RenderHandlerThread()
+    mockkStatic("com.mapbox.maps.MapboxLogger")
```
Comment (alexander-kulikovskii) [inline sdk/src/test/java/com/mapbox/maps/renderer/RenderHandlerThreadTest.kt:24]: should we add `unmockkStatic`?

File: sdk-base/src/main/java/com/mapbox/maps/MapboxLogger.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
+@file:JvmName("MapboxLogger")
+
+package com.mapbox.maps
+
+import com.mapbox.common.Logger
```
Comment (ank27) [inline sdk-base/src/main/java/com/mapbox/maps/MapboxLogger.kt:5]: isn't this deprecated in common?

Comment (kiryldz) [reply sdk-base/src/main/java/com/mapbox/maps/MapboxLogger.kt:5]: it is but we should still use it, no replacement yet 😉

File: sdk-base/src/main/java/com/mapbox/maps/MapboxLogger.kt [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * Log Android error entry in the following format: E/Mapbox: [[SDK_IDENTIFIER]\ [tag]] [message]
+ */
+fun logE(tag: String, message: String) = Logger.e("$SDK_IDENTIFIER\\$tag", message)
```
Comment (yunikkk) [inline sdk-base/src/main/java/com/mapbox/maps/MapboxLogger.kt:10]: So previously it was plain `tag`, now tag is `maps-android/tag`, where is `Mapbox` originating from? Is it inserted in common?

Comment (kiryldz) [reply sdk-base/src/main/java/com/mapbox/maps/MapboxLogger.kt:10]: you're totally correct

File: sdk/src/test/java/com/mapbox/maps/renderer/RenderHandlerThreadTest.kt (hunk 5 above)
Comment (kiryldz) [reply sdk/src/test/java/com/mapbox/maps/renderer/RenderHandlerThreadTest.kt:25]: done across all the tests

**Review by alexander-kulikovskii — APPROVED** [review]

**Review by yunikkk — APPROVED** [review]

**Pull request merged**